date-time/src/commonMain/kotlin/com/splendo/kaluga/datetime/timer/RecurringTimer.kt

```diff
@@ -52,7 +52,13 @@ class RecurringTimer(
     coroutineScope: CoroutineScope = MainScope()
 ) : ControllableTimer {
     private val stateRepo = TimerStateRepo(duration, interval, timeSource, delayFunction, coroutineScope)
-    override val state: StateFlow<Timer.State> = stateRepo.stateFlow
+    override val state: Flow<Timer.State> = stateRepo.stateFlow.map { state ->
+        when (state) {
+            is TimerStateRepo.State.NotRunning.Finished -> state
+            is TimerStateRepo.State.NotRunning.Paused -> state
+            is TimerStateRepo.State.Running -> state
+        }
+    }
```

Comment:
Not sure I understand the mapping.

Also changing `StateFlow` to `Flow` can generally be good if there is no meaningful start state, but I don't think that is the case here?

Reply:
Hm, it's interesting the compiler doesn't want to handle this, `TimeStateRepo.State` is sealed and all subclasses are a `Timer.State`

`as StateFlow<Timer.State>` does work in tests but is of course a warning, and someone could make a non Timer.State subclass one day. However, any use of this state would result in a crash.

Do you prefer the current solution? If so we should just add a comment why.

Reply:
It's error prone, but the error would indicate we have something wrong in our implementation. Of course the `when` would also indicate something is wrong, and compile time so it is better at that (just a comment to explain why it exists would be good).

However we of course loose the initial state. This seems rather important to me for e.g. making an `InitializedObservable`.

We can remedy this by exposing is separately, or making a suspended method to do the stateIn (or both with the latter as an extension method).

Reply:
decided to just go with exposing the state seperatly.